brave/src/main/java/brave/handler/MutableSpan.java

```diff
@@ -81,9 +85,9 @@
   int localPort, remotePort;
 
   /** To reduce the amount of allocation use a pair-indexed list for tag (key, value). */
-  ArrayList<String> tags;
-  /** Also use pair indexing for annotations, but type object to store (startTimestamp, value). */
-  ArrayList<Object> annotations;
+  Object[] tags = EMPTY_ARRAY;
```

Comment:
Just to confirm it's ok for this to not be volatile. I guess an `ArrayList` would have already had issues when resizing though if there is a problem, but I'm not sure that memory is synced if users allowed to mutate without the lock.

Reply:
yeah this type isn't thread safe it is externally synchronized. the only reason doing copy-on-write is for the map views, to avoid having issues with people sharing them. I'll make a note

Reply:
copy-on-write for update I mean

Reply:
I think I added confusion by even saying copy-on-write.. my bad. PTAL at latest!